pkg/k8s/watchers/cilium_endpoint.go

```diff
@@ -29,87 +24,73 @@ import (
-			&cilium_v2.CiliumEndpoint{},
-			0,
-			cache.ResourceEventHandlerFuncs{
-				AddFunc: func(obj interface{}) {
-					var valid, equal bool
-					defer func() {
-						k.K8sEventReceived(apiGroup, metricCiliumEndpoint, resources.MetricCreate, valid, equal)
-					}()
-					if ciliumEndpoint, ok := obj.(*types.CiliumEndpoint); ok {
-						valid = true
-						k.endpointUpdated(nil, ciliumEndpoint)
-						k.K8sEventProcessed(metricCiliumEndpoint, resources.MetricCreate, true)
-					}
-				},
-				UpdateFunc: func(oldObj, newObj interface{}) {
-					var valid, equal bool
-					defer func() { k.K8sEventReceived(apiGroup, metricCiliumEndpoint, resources.MetricUpdate, valid, equal) }()
-					if oldCE := k8s.CastInformerEvent[types.CiliumEndpoint](oldObj); oldCE != nil {
-						if newCE := k8s.CastInformerEvent[types.CiliumEndpoint](newObj); newCE != nil {
-							valid = true
-							if oldCE.DeepEqual(newCE) {
-								equal = true
-								return
-							}
-							k.endpointUpdated(oldCE, newCE)
-							k.K8sEventProcessed(metricCiliumEndpoint, resources.MetricUpdate, true)
-						}
-					}
-				},
-				DeleteFunc: func(obj interface{}) {
-					var valid, equal bool
-					defer func() { k.K8sEventReceived(apiGroup, metricCiliumEndpoint, resources.MetricDelete, valid, equal) }()
-					ciliumEndpoint := k8s.CastInformerEvent[types.CiliumEndpoint](obj)
-					if ciliumEndpoint == nil {
-						return
-					}
-					valid = true
-					k.endpointDeleted(ciliumEndpoint)
-				},
-			},
-			k8s.TransformToCiliumEndpoint,
-			cache.Indexers{
-				"localNode": CreateCiliumEndpointLocalPodIndexFunc(),
-			},
+		var synced atomic.Bool
+		stop := make(chan struct{})
+
+		k.blockWaitGroupToSyncResources(
+			stop,
+			nil,
+			func() bool { return synced.Load() },
+			apiGroup,
 		)
-		k.ciliumEndpointIndexerMU.Lock()
-		k.ciliumEndpointIndexer = cepIndexer
-		k.ciliumEndpointIndexerMU.Unlock()
-		isConnected := make(chan struct{})
-		// once isConnected is closed, it will stop waiting on caches to be
-		// synchronized.
-		k.blockWaitGroupToSyncResources(isConnected, nil, ciliumEndpointInformer.HasSynced, k8sAPIGroupCiliumEndpointV2)
-
-		once.Do(func() {
-			// Signalize that we have put node controller in the wait group
-			// to sync resources.
-			asyncControllers.Done()
-		})
-		k.k8sAPIGroups.AddAPI(k8sAPIGroupCiliumEndpointV2)
-		go ciliumEndpointInformer.Run(isConnected)
-
-		<-kvstore.Connected()
-		close(isConnected)
-
-		log.Info("Connected to key-value store, stopping CiliumEndpoint watcher")
+		k.k8sAPIGroups.AddAPI(apiGroup)
+
+		// Signalize that we have put node controller in the wait group to sync resources.
+		once.Do(asyncControllers.Done)
+
+		// derive another context to signal Events() in case of kvstore connection
+		eventsCtx, cancel := context.WithCancel(ctx)
+
+		go func() {
+			defer close(stop)
+
+			events := k.resources.CiliumSlimEndpoint.Events(eventsCtx)
+			cache := make(map[resource.Key]*types.CiliumEndpoint)
```

Comment:
Here as well in the other related PR I would again argue that we shouldn't maintain this cache thing nor do DeepEqual etc. checks. But let's tackle that separately.